HW2/P2/P2.txt

```diff
@@ -0,0 +1,12 @@
+
+
+Characterize the performance of each approach (coarse, medium, and fine-grained lock- ing) on the two cases in P2.py: random data exchanges and correlated data exchanges.
+
+ I executed experiments under 6 different combinations of lock refinement and found the serial versions to run much faster than any of the multi threaded version. I believe this is because this function employs numpy type arrays which have been optimized computationally. Of the threaded versions the uncorrleated medium grained version appeared to run the fastest. I believe this is because the process is not overburdened by logic of the locks. When considering the number of threads to be used, performance appears to be optimized at about 4 threads for all multi threaded versions of the function. I would suggest a thread count of 4 for running these functions. Please see the plot: "P2_plot_threads.png" for all data. 
```

Comment:
Good point. I never thought of the optimization in numpy as a factor. Did the TF confirm this?